catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -662,46 +662,40 @@ private Metacard generateMetacard(String mimeTypeRaw, String id, String fileName
             throws MetacardCreationException, MimeTypeParseException {
 
         Metacard generatedMetacard = null;
-        InputTransformer transformer = null;
-        StringBuilder causeMessage = new StringBuilder("Could not create metacard with mimeType ");
-        try {
-            MimeType mimeType = new MimeType(mimeTypeRaw);
+        InputTransformer transformer;
+        StringBuilder causeMessage = new StringBuilder();
 
-            List<InputTransformer> listOfCandidates =
-                    frameworkProperties.getMimeTypeToTransformerMapper()
-                            .findMatches(InputTransformer.class, mimeType);
+        MimeType mimeType = new MimeType(mimeTypeRaw);
 
-            LOGGER.debug("List of matches for mimeType [{}]: {}", mimeType, listOfCandidates);
+        List<InputTransformer> listOfCandidates = frameworkProperties.getMimeTypeToTransformerMapper().findMatches(InputTransformer.class, mimeType);
 
-            for (InputTransformer candidate : listOfCandidates) {
-                transformer = candidate;
+        LOGGER.debug("List of matches for mimeType [{}]: {}", mimeType, listOfCandidates);
 
-                try (InputStream transformerStream = com.google.common.io.Files.asByteSource(
-                        tmpContentPath.toFile())
-                        .openStream()) {
-                    generatedMetacard = transformer.transform(transformerStream);
-                }
-                if (generatedMetacard != null) {
-                    break;
+        for (InputTransformer candidate : listOfCandidates) {
+            transformer = candidate;
+
+            try (InputStream transformerStream = com.google.common.io.Files.asByteSource(tmpContentPath.toFile()).openStream()) {
+                generatedMetacard = transformer.transform(transformerStream);
+            } catch (CatalogTransformerException | IOException e) {
+                causeMessage.append(String.format("Transformer [%s] could not create metacard.", transformer));
+                /*
+                    The caught exception more than likely does not have the root cause message
+                    that is needed to inform the caller as to why things have failed.  Therefore
+                    we need to iterate through the chain of cause exceptions and gather up
+                    all of their message details.
+                */
+                causeMessage.append(mimeTypeRaw).append(". Reason: ").append(System.lineSeparator()).append(e.getMessage());
```

Comment:
I am not personally a fan of doing this level of log/exception message formatting. In this case, I can certainly see the reason for doing so, but I'd much rather it be extracted out of the business logic.

By maintaining a List<Throwable>, you could add an overloaded ctor to `MetacardCreationException` that would take `MetacardCreationException(String msg, String mimeTypeRaw, List<Throwable> exceptions)` and manage its own string processing internally. 

This would clean up your business logic to the bare minimum - try; catch; add exception to list; print debug log - and isolate the formatting of the exception to the class best suited to it.

It would also make it easier to avoid appending `mimeTypeRaw` to the message on every failure.